### PR TITLE
Added centralized config resolution and utilization for Scan-PR flow

### PR DESCRIPTION
**Pull request by eranturgeman**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
- [x] Update [documentation](https://github.com/jfrog/documentation) about new features / new supported technologies
---

**IMPORTANT NOTE:** This new capability is currently intended for use by JFrog internal processes ONLY! It will not be supported for customers until the official release!

This PR introduces the integration of a new config profile into the Frogbot Scan-PR flow.
**Note:** Currently, we only support a single-module profile with a '.' as the path_from_root. Profiles with different settings will be rejected, causing the flow to fail.

As an initial integration step, the only fields considered from the config profile, if provided, are enable_sast_scan and enable_secrets_scan. All other scan-related configurations in the profile **will be ignored**.

When a config profile is used for the scan, only the SAST and Secrets scanners can be activated. Other requested scanners, such as SCA, IaC, and CA, **will not be executed**.

If a config profile is provided, it **takes priority over the jfrog-apps-config.yml** file.

### CODE REVIEW

File: utils/consts.go [hunk 1]
```diff
@@ -31,6 +31,7 @@ const (
 	jfrogReleasesRepoEnv   = "JF_RELEASES_REPO"
 	JFrogPasswordEnv       = "JF_PASSWORD"
 	JFrogTokenEnv          = "JF_ACCESS_TOKEN"
+	JfrogConfigProfileEnv  = "JF_CONFIG_PROFILE" // JFROG internal use ONLY!
```
Comment (hadarshjfrog) [inline utils/consts.go:34]: I'd remove the comment. It won't be in the official documentation until it's GA for all.

Comment (attiasas) [reply utils/consts.go:34]: Also comments should be above the line and inline should be avoided

File: utils/params.go [hunk 2]
```diff
@@ -706,3 +720,30 @@ func readConfigFromTarget(client vcsclient.VcsClient, gitParamsFromEnv *Git) (co
...
+		if configProfile.Modules[0].PathFromRoot != "." {
+			err = fmt.Errorf("module '%s' in profile '%s' contains the following path from root: '%s'. Frogbot currently supports only a single module with a '.' path from root", configProfile.Modules[0].ModuleName, profileName, configProfile.Modules[0].PathFromRoot)
+			return
+		}
+		log.Info(fmt.Sprintf("Config profile '%s' was found. All scanners configurations will be featched from env vars and the config profile. jfrog-apps-config will be ignored if exists", profileName))
+		log.Info("Utilizing config profile is currently intended for Jfrog internal use ONLY. If used- the only supported feature is enablement/exclusion of Secrets and Sast scanners")
+	} else {
+		log.Info("No JF_CONFIG_PROFILE environment variable was provided. All configurations will be induced from Env vars and files")
+	}
+	return
```
Comment (attiasas) [inline utils/params.go:748]:
```suggestion
	profileName := getTrimmedEnv(JfrogConfigProfileEnv)
    if profileName == "" {
        log.Debug(fmt.Sprintf("No %s environment variable was provided. All configurations will be induced from Env vars and files"), JfrogConfigProfileEnv)
        return
    }
	if configProfile, err = xsc.GetConfigProfile(jfrogServer, profileName); err != nil {
		return
	}
    // Currently, only a single Module that represents the entire project is supported
	if len(configProfile.Modules) != 1 {
		err = fmt.Errorf("more than one module was found '%s' profile. Frogbot currently supports only one module per config profile", configProfile.ProfileName)
		return
	}
    if configProfile.Modules[0].PathFromRoot != "." {
		err = fmt.Errorf("module '%s' in profile '%s' contains the following path from root: '%s'. Frogbot currently supports only a single module with a '.' path from root", configProfile.Modules[0].ModuleName, profileName, configProfile.Modules[0].PathFromRoot)
		return
	}
    log.Info(fmt.Sprintf("Config profile '%s' was found. All scanners configurations will be featched from env vars and the config profile. jfrog-apps-config will be ignored if exists", profileName))
	log.Info("Utilizing config profile is currently intended for Jfrog internal use ONLY. If used, the only supported feature is enablement/exclusion of Secrets and Sast scanners")
	return
```
* Keep code flat if possible
* I think logs about this should be in Debug unless found and used.
* Make sure the last log content is format, clear and approved

File: utils/params_test.go [hunk 3]
```diff
@@ -678,3 +681,53 @@ func TestSetEmailDetails(t *testing.T) {
...
+		profileName     string
+		failureExpected bool
+	}{
+		/*
+			{
+				profileName:     ValidConfigProfile,
+				failureExpected: false,
+			},
+
+		*/
```
Comment (attiasas) [inline utils/params_test.go:696]: commented out tests, please add them

File: utils/testsutils.go [hunk 4]
```diff
@@ -52,6 +63,22 @@ func SetEnvsAndAssertWithCallback(t *testing.T, envs map[string]string) func() {
...
+	if exist {
+		return func() {
+			setEnvAndAssert(t, key, oldValue)
+		}
+	}
+
+	return func() {
+		unsetEnvAndAssert(t, key)
+	}
+}
```
Comment (attiasas) [inline utils/testsutils.go:80]: we already have implementation for this at `clientTests.SetEnvWithCallbackAndAssert`
remove this

File: utils/testsutils.go [hunk 5]
```diff
@@ -142,3 +169,64 @@ func CreateTempJfrogHomeWithCallback(t *testing.T) (string, func()) {
...
+			updatedContent = strings.Replace(updatedContent, `"path_from_root": "."`, `"path_from_root": "backend"`, 1)
+
+			_, err = w.Write([]byte(updatedContent))
+			assert.NoError(t, err)
+		case r.RequestURI == "/xsc/api/v1/system/version":
+			_, err := w.Write([]byte(fmt.Sprintf(`{"xsc_version": "%s"}`, services.ConfigProfileMinXscVersion)))
+			assert.NoError(t, err)
+		default:
+			assert.Fail(t, "received an unexpected request")
+		}
```
Comment (attiasas) [inline utils/testsutils.go:222]:
```suggestion
        const secondModule = services.Module{
			ModuleId:     999,
			ModuleName:   "second-module",
			PathFromRoot: ".",
			ScanConfig: services.ScanConfig{
				ScanTimeout:                  0,
				ExcludePattern:               "",
				EnableScaScan:                false,
				EnableContextualAnalysisScan: false,
            },
		},
        switch {
        case strings.HasPrefix(r.RequestURI, "/xsc/api/v1/profile/"):
            assert.Equals(t, http.MethodGet, r.Method)
            if r.RequestURI == "/xsc/api/v1/profile/"+ValidConfigProfile {
                w.WriteHeader(http.StatusOK)
            } else {
                w.WriteHeader(http.StatusBadRequest)
            }
            content, err := os.ReadFile("../testdata/configprofile/configProfileExample.json")
			assert.NoError(t, err)

            if r.RequestURI == "/xsc/api/v1/profile/"+InvalidModulesConfigProfile {
                // Adding a second module to make the profile invalid, as we currently support ONLY profile with a single module
                var profile services.ConfigProfile
                err = json.Unmarshal(content, &profile)
                assert.NoError(t, err)
                profile.Modules = append(profile.Modules, secondModule)
                content, err = json.Marshal(profile)
                assert.NoError(t, err)
            }

            if r.RequestURI == "/xsc/api/v1/profile/"+InvalidPathConfigProfile {
                // Changing 'path_from_root' to a path different from '.' to make the module invalid, as we currently support ONLY a single module with '.' path
                updatedContent := string(content)
                updatedContent = strings.Replace(updatedContent, `"path_from_root": "."`, `"path_from_root": "backend"`, 1)
            }

            _, err = w.Write(content)
			assert.NoError(t, err)
        case r.RequestURI == "/xsc/api/v1/system/version":
			_, err := w.Write([]byte(fmt.Sprintf(`{"xsc_version": "%s"}`, services.ConfigProfileMinXscVersion)))
			assert.NoError(t, err)
		default:
			assert.Fail(t, "received an unexpected request")
        }
```
* Remove code duplications if possiable

**Review by attiasas — CHANGES_REQUESTED** [review]
Nice, take a look at my comments

**Pull request merged**